src/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.cs

```diff
@@ -356,6 +356,9 @@ public sealed override MemberTypes MemberType
             }
         }
 
+        //
+        // Left unsealed as there are so many subclasses. Need to be overriden by EcmaFormatRuntimeNamedTypeInfo and RuntimeConstructedGenericTypeInfo
+        //
```

Comment:
Just did a quick experiment on the full framework: MetadataToken on "haselement" types (array, pointer, byref) seems to return the MetadataToken of the element type.

Maybe the "abstract override" discipline is worth it here too (even if only an experiment to force visiting all these cases.)